node/peer.js

```diff
@@ -149,13 +149,20 @@ TChannelPeer.prototype.getInConnection = function getInConnection() {
     return null;
 };
 
-TChannelPeer.prototype.getOutConnection = function getOutConnection() {
+TChannelPeer.prototype.getOutConnection = function getOutConnection(identified) {
     var self = this;
+    var candidate = null;
     for (var i = self.connections.length - 1; i >= 0; i--) {
         var conn = self.connections[i];
-        if (!conn.closing) return conn;
+        if (!conn.closing) {
+            if (!identified || conn.remoteName) {
```

Comment:
:+1:  please let's do:
```
if (conn.closing) continue;
if (!identified) return conn; // user doesn't care, take last outgoing
if (conn.remoteName) return conn; // user wanted an identified channel, and we found one
if (!candidate) candidate = conn; // we'll fallback to returning this if we can't find an identified one
```

Also please `s/identified/preferIdentified/`, since I'd infer otherwise that if I set `identified` to true, it's a guarantee.